src/workerd/jsg/setup.h

```diff
@@ -485,12 +485,19 @@ class Isolate: public IsolateBase {
     // executing JavaScript code. T should be one of your API types which you want to use as the
     // global object. `args...` are passed to the type's constructor.
     template <typename T, typename... Args>
-    JsContext<T> newContext(Args&&... args) {
+    JsContext<T> newContext(NewContextOptions options, Args&&... args) {
       // TODO(soon): Requiring move semantics for the global object is awkward. This should instead
       //   allocate the object (forwarding arguments to the constructor) and return something like
       //   a Ref.
+      return jsgIsolate.wrapper->newContext(*this, options, jsgIsolate.getObserver(), (T*)nullptr, kj::fwd<Args>(args)...);
+    }
 
-      return jsgIsolate.wrapper->newContext(*this, jsgIsolate.getObserver(), (T*)nullptr, kj::fwd<Args>(args)...);
+    // Creates a new JavaScript "context", i.e. the global object. This is the first step to
+    // executing JavaScript code. T should be one of your API types which you want to use as the
+    // global object. `args...` are passed to the type's constructor.
```

Comment:
Nit: comment is repeated from above and perhaps redundant, this can stay as-is if you think that this is more clear.

Reply:
Since the comments are documenting two different methods, I think there's value in keep them, even if redundant